packages/generator-common/src/options.ts

```diff
@@ -0,0 +1,171 @@
+
+/**
+ * @internal
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function getCommonCliOptions(readmeText: string) {
+  return {
+    optionsPerService: {
+      alias: 's',
+      describe:
+        'Configuration file to ensure consistent names between multiple generation runs with updated / changed metadata files. The configuration allows to set a `directoryName` and `npmPackageName` for every service, identified by the path to the original file. It also makes sure that names do not change between generator runs. If a directory is passed, a `options-per-service.json` file is read/created in this directory.',
+      type: 'string',
+      coerce: resolveOptionsPerService
+    },
+    clearOutputDir: {
+      type: 'boolean',
+      describe:
+        'Remove all files in the output directory before generation. Be cautious when using this option, as it really removes EVERYTHING in the output directory.',
+      default: false
+    },
+    include: {
+      type: 'string',
+      coerce: resolveGlob,
+      describe:
+        'Include files matching the given glob into the root of each generated client directory.',
+      requiresArg: true
+    },
+    metadata: {
+      describe: 'When set to true, SDK metadata for the API hub is generated.',
+      type: 'boolean',
+      default: false,
+      hidden: true
+    },
+    outputDir: {
+      type: 'string',
+      alias: 'o',
+      describe:
+        'Specify the path to the directory to generate the client(s) in. Each client is generated into a subdirectory within the given output directory. Creates the directory if it does not exist. Customize subdirectory naming through `--optionsPerService`.',
+      coerce: resolveRequiredPath,
+      demandOption: true,
+      requiresArg: true
+    },
+    overwrite: {
+      describe:
+        'By default, the generator will exit when encountering a file that already exists. When set to true, it will be overwritten instead. Please note that compared to the --clearOutputDir option, this will not delete outdated files.',
+      type: 'boolean',
+      default: false
+    },
+    packageJson: {
+      describe:
+        'When enabled, a `package.json` that specifies dependencies and scripts for transpilation is generated.',
+      type: 'boolean',
+      default: false
+    },
+    prettierConfig: {
+      type: 'string',
+      alias: 'p',
+      describe:
+        'Specify the path to the prettier config. If not given a default config will be used for the generated sources.',
+      coerce: resolvePath,
+      requiresArg: true
+    },
+    skipValidation: {
+      type: 'boolean',
+      describe:
+        'By default, the generation fails, when there are duplicate or invalid names for operations and/or path parameters after transforming them to camel case. Set this to true to enable unique and valid name generation. The names will then be generated by appending numbers and prepending prefixes.',
```

Comment:
I feel path parameters are specific to OpenAPI. Maybe we can adjust it so it works for both